lib/mix/lib/mix.ex

```diff
@@ -746,6 +802,16 @@ defmodule Mix do
     end
   end
 
+  defp install_dir(cache_id) do
+    # TODO: rename MIX_INSTALL_DIR to MIX_INSTALL_ROOT?
```

Comment:
👀 

Notice that I've added a `__install_dir__` option which is _really_ useful for testing. And so in hindsight I think _dir is a better suffix for the actual dir and _root for the cache root.